File: app/src/main/java/com/simplemobiletools/calendar/receivers/BootCompletedReceiver.java
package com.simplemobiletools.calendar.receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.simplemobiletools.calendar.DBHelper;
import com.simplemobiletools.calendar.Utils;
import com.simplemobiletools.calendar.models.Event;

import java.util.List;

public class BootCompletedReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent arg1) {
        final List<Event> events = DBHelper.newInstance(context, null).getEventsAtReboot();
        for (Event event : events) {
            Utils.scheduleNotification(context, event);
        }
    }
}

File: app/src/main/kotlin/com/simplemobiletools/calendar/receivers/BootCompletedReceiver.kt
package com.simplemobiletools.calendar.receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.simplemobiletools.calendar.extensions.dbHelper
import com.simplemobiletools.calendar.extensions.notifyRunningEvents
import com.simplemobiletools.calendar.extensions.recheckCalDAVCalendars
import com.simplemobiletools.calendar.extensions.scheduleAllEvents

class BootCompletedReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        // try just getting a reference to the db so it updates in time
        context.dbHelper

        Thread {
            context.apply {
                scheduleAllEvents()
                notifyRunningEvents()
                recheckCalDAVCalendars {}
            }
        }.start()
    }
}
